### PR TITLE
Sync entries with medications/contacts

### LINKED ISSUE #44: updates to entry form (selection dropdown, scale number, date input form, etc.)
potentially same fix as #32 

- [x] Default date (today) for create form
- [x] dropdown for type, condition
- [x] dropdown for detail after sync?
- [x] number input for scale (1-10) --> moved to #84 
- [x] potential date selector (choose from calendar) in form
- [x] auto-populate type field
- [x] change detail field label based on type selection

### LINKED ISSUE #61: sync between entry and linked concepts
- [x] entry <--> medication
- [x] entry <--> contact

### LINKED ISSUE #86: navlink wrapped in button

### PR DESCRIPTION
**Pull request by jrdegreeff**
closes #61
closes #44
closes #86
partially addresses #87

### CODE REVIEW

**Review by oliviagiandrea — DISMISSED** [review]
Looks good to me. Took me a while to understand some bits, but that's due to my own inexperience, not the quality of your code. Great job!

File: server/entry/middleware.ts [hunk 1]
```diff
@@ -35,6 +52,39 @@ const isValidEntryType = (required: boolean) => async (req: Request, res: Respon
...
+    if (!record) {
+        res.status(404).json({
+            error: `${types[type].detailName} with id ${req.body.detail} does not exist.`
+        });
+        return;
+    }
+
+    if (req.session.accountId !== record.owner._id.toString()) {
+        console.log(req.session.accountId)
+        console.log(record.owner._id.toString())
```
Comment (wli-linda) [inline server/entry/middleware.ts]: nit

File: client/components/Entry/CreateEntryForm.vue [hunk 2]
```diff
@@ -17,13 +17,39 @@ export default {
...
+        type: form => {
+          const detailTypes = {
+            '': {id: 'detail', label: '', type: 'hidden'},
+            appointment: {id: 'detail', label: 'doctor', type: 'select', options: this.$store.getters.activeContacts.map(
+              c => { return {value: c._id, text: c.title}; }
+            )},
+            medication: {id: 'detail', label: 'medication', type: 'select', options: this.$store.getters.activeMedications.map(
+              m => { return {value: m._id, text: m.title}; }
+            )},
+            other: {id: 'detail', label: 'description'},
```
Comment (wli-linda) [inline client/components/Entry/CreateEntryForm.vue]: nit: capitalize Doctor, Medication, Description

Comment (wli-linda) [reply client/components/Entry/CreateEntryForm.vue]: also, "doctor"? or maybe "healthcare provider"? "appointment with"?

File: client/components/Entry/EntryComponent.vue [hunk 3]
```diff
@@ -131,4 +127,8 @@ td {
     width: 8rem;
     margin: 0rem 1rem;
 }
+
+td > a {
+    text-decoration: underline;
+}
```
Comment (wli-linda) [inline client/components/Entry/EntryComponent.vue:133]: nice

**Review by wli-linda — COMMENTED** [review]
Saw this but wasn't sure where to fix it in the current implementation:
image
image

i.e. when fields are prepopulated, detail input doesn't show up
but let's merge the PR for now? focus on your other project, I can take a harder look / we can just let it be

**Review by wli-linda — APPROVED** [review]

**Pull request merged**